src/main/java/net/coderbot/iris/uniforms/SamplerUniforms.java

```diff
@@ -32,7 +33,7 @@ private SamplerUniforms() {
 		// no construction allowed
 	}
 
-	public static void addCommonSamplerUniforms(UniformHolder uniforms) {
+	public static void addCommonSamplerUniforms(LocationalUniformHolder uniforms) {
```

Comment:
Does this need to be a locational uniform holder, feels like UniformHolder work be acceptable here too?

Reply:
The samplers need the `location` to select the correct `shadow` sampler. It changes depending on whether the `watershadow` is used tr not it seems.

Reply:
Right, forgot about that, in that case this is fine for now. Might be worth adding comments about that though.